### ISSUE TITLE
Cakephp 4.x - issue with utf8 characters 

### ISSUE THREAD

**Issue by ttsraja** [cakephp/cakephp#14430]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

```
'Datasources' => [
        'default' => [
            'host' => 'localhost',
            /*
             * CakePHP will use the default DB port based on the driver selected
             * MySQL on MAMP uses port 8889, MAMP users will want to uncomment
             * the following line and set the port accordingly
             */
            //'port' => 'non_standard_port_number',
            'username' => 'root',
            'password' => 'root',
            'database' => 'tsapp,
            'encoding' => 'utf8mb4',
            //'flags' => [\PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8'],
            /**
             * If not using the default 'public' schema with the PostgreSQL driver
             * set it here.
             */
            //'schema' => 'myapp',

            /**
             * You can use a DSN string to set the entire configuration
             */
            'url' => env('DATABASE_URL', null),
        ],
```

I have this in my app_local.php and below in my app.php
```
'default' => [
            'className' => Connection::class,
            'driver' => Mysql::class,
            'persistent' => false,
            'timezone' => 'UTC',
            /**
             * For MariaDB/MySQL the internal default changed from utf8 to utf8mb4, aka full utf-8 support, in CakePHP 3.6
             */
            'encoding' => 'utf8mb4',

            /**
             * If your MySQL server is configured with `skip-character-set-client-handshake`
             * then you MUST use the `flags` config to set your charset encoding.
             * For e.g. `'flags' => [\PDO::MYSQL_ATTR_INIT_COMMAND => 'SET NAMES utf8mb4']`
             */
            'cacheMetadata' => true,
            'log' => true,
            /*
             * Set identifier quoting to true if you are using reserved words or
             * special characters in your table or column names. Enabling this
             * setting will result in queries built using the Query Builder having
             * identifiers quoted when creating SQL. It should be noted that this
             * decreases performance because each query needs to be traversed and
             * manipulated before being executed.
             */
            'quoteIdentifiers' => true,

            /*
             * During development, if using MySQL < 5.6, uncommenting the
             * following line could boost the speed at which schema metadata is
             * fetched from the database. It can also be set directly with the
             * mysql configuration directive 'innodb_stats_on_metadata = 0'
             * which is the recommended value in production environments
             */
            //'init' => ['SET GLOBAL innodb_stats_on_metadata = 0'],
        ],

```

I have utf8mb4_general_ci collation in my tables, database and everywhere possible. but still I couldnt achieve what I am supposed to get.

It was working fine in Cakephp 1.x version. I am in process of migrating the site. I have tried import from production server & export to the localhost without changing anything. But still I couldnt make as work in production.
Unicode text are not properly displaying the view. Its viewing as à®šà¯†à®©à¯à®©à¯ˆ

**Comment by othercorey** [comment]
- What version of mysql are you using?
- Can you show an example of a table schema?
- Can you show an example of displaying the string?

**Comment by ttsraja** [comment]
I am using MySQL - 5.7.25 

Update : I managed to make it work by changing the encoding in the database configuration
```
`
'encoding' => 'latin1'

`
```

I found data was inserted with latin1_general_ci and it was working fine with the current production running with Cakephp v1.x. Now after making the above changes I could able to insert and view data's properly. I haven't changed the table collation or database collation it still have **utf8mb4** only. It working fine.

Thanks all for your time.
